### PR TITLE
Encode revision IDs used in replication apis

### PR DESCRIPTION
**Pull request by danielhlarkin**
### Scope & Purpose

The existing APIs for the new (but still deactivated) replication protocol simply exposed `_rev` values as numbers, which is not fully portable, as JS actually only supports numbers up to about 2^53, and we can exceed that with our values. This PR fixes this by encoding all of these values as strings in the APIs which transport the values.

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as *(please describe tests)*.

Jenkins, protocol enabled: http://172.16.10.101:8080/job/arangodb-matrix-pr/8921/ (all blue)
Jenkins, protocol disabled: http://172.16.10.101:8080/job/arangodb-matrix-pr/8924/

### CODE REVIEW

File: arangod/RestHandler/RestReplicationHandler.cpp [hunk 1]
```diff
@@ -3107,45 +3115,52 @@ void RestReplicationHandler::handleCommandRevisionRanges() {
   {
     VPackObjectBuilder obj(&response);
     TRI_voc_rid_t resumeNext = ctx.resume;
-    VPackSlice range = body.at(current);
+    VPackSlice range;
+    std::size_t left;
```
Comment (neunhoef) [inline arangod/RestHandler/RestReplicationHandler.cpp]: Ii suggest to use `uint64_t` throughout for revs. We even have `TRI_voc_rid_t` which is `uint64_t`, even on 32bit machines! `size_t` on 32bit machines will be 32bit only, so this will probably not even compile without warnings on Windows at least. Let's change things here to `uint64_t` and then do another PR to change `size_t` in lots of places (including the MerkleTree files).

File: arangod/Replication/DatabaseInitialSyncer.cpp [hunk 2]
```diff
@@ -1493,15 +1501,19 @@ Result DatabaseInitialSyncer::fetchCollectionSyncByRevisions(arangodb::LogicalCo
...
-        std::size_t removalBound = masterSlice.isEmptyArray()
-                                       ? currentRange.second + 1
-                                       : masterSlice.at(0).getNumber<std::size_t>();
+        std::size_t removalBound =
+            masterSlice.isEmptyArray()
+                ? currentRange.second + 1
+                : basics::HybridLogicalClock::decodeTimeStamp(masterSlice.at(0));
+        if (currentRange.first > removalBound) {
+          LOG_DEVEL << currentRange.first << " > " << removalBound;
+        }
```
Comment (neunhoef) [inline arangod/Replication/DatabaseInitialSyncer.cpp]: Debug logging remnant.

File: arangod/Replication/DatabaseInitialSyncer.cpp [hunk 3]
```diff
@@ -1460,14 +1468,15 @@ Result DatabaseInitialSyncer::fetchCollectionSyncByRevisions(arangodb::LogicalCo
...
+      if (!resumeSlice.isNone() && !resumeSlice.isString()) {
         return Result(TRI_ERROR_REPLICATION_INVALID_RESPONSE,
                       std::string("got invalid response from master at ") +
                           _config.master.endpoint + batchUrl +
                           ": response field 'resume' is not a number");
       }
-      requestResume = resumeSlice.isNone() ? std::numeric_limits<std::size_t>::max()
-                                           : resumeSlice.getNumber<std::size_t>();
+      requestResume = resumeSlice.isNone()
+                          ? std::numeric_limits<std::size_t>::max()
```
Comment (neunhoef) [inline arangod/Replication/DatabaseInitialSyncer.cpp]: This should be `uint64_t` instead of `size_t`. See below.

File: arangod/Replication/DatabaseInitialSyncer.cpp [hunk 4]
```diff
@@ -1493,15 +1501,19 @@ Result DatabaseInitialSyncer::fetchCollectionSyncByRevisions(arangodb::LogicalCo
                               static_cast<TRI_voc_rid_t>(currentRange.first)));
         }
 
-        std::size_t removalBound = masterSlice.isEmptyArray()
-                                       ? currentRange.second + 1
-                                       : masterSlice.at(0).getNumber<std::size_t>();
+        std::size_t removalBound =
```
Comment (neunhoef) [inline arangod/Replication/DatabaseInitialSyncer.cpp]: uint64_t

File: arangod/RestHandler/RestReplicationHandler.cpp [hunk 5]
```diff
@@ -3071,23 +3072,30 @@ void RestReplicationHandler::handleCommandRevisionRanges() {
...
       VPackSlice first = entry.at(0);
       VPackSlice second = entry.at(1);
-      if (!first.isNumber() || !second.isNumber()) {
+      if (!first.isString() || !second.isString()) {
+        badFormat = true;
+        break;
+      }
+      std::uint64_t left = basics::HybridLogicalClock::decodeTimeStamp(first);
+      std::uint64_t right = basics::HybridLogicalClock::decodeTimeStamp(second);
+      if (left == std::numeric_limits<std::size_t>::max() ||
```
Comment (neunhoef) [inline arangod/RestHandler/RestReplicationHandler.cpp]: uint64_t

File: arangod/Replication/DatabaseInitialSyncer.cpp (hunk 2 above)
Comment (danielhlarkin) [reply arangod/Replication/DatabaseInitialSyncer.cpp]: Sorry, thanks for catching that.

**Comment by danielhlarkin** [comment]
http://172.16.10.101:8080/job/arangodb-matrix-pr/8932/

**Comment by danielhlarkin** [comment]
http://172.16.10.101:8080/job/arangodb-matrix-pr/8942/ (all blue)
